content/en/docs/contribution-guidelines/upgrade-graphs.md

```diff
@@ -0,0 +1,241 @@
+---
+title: Channel Upgrade Graphs Legends
+linkTitle: Upgrade Graphs Legends
+description: >
+  Defines the legends for channel upgrade graph representation schemes used. 
+---
+
+By checking this document, you will be able to understand the legends used in the Channel Upgrade Graphs as how to create them. Note that it also can be useful to drawing your own graphs to represent your scenarios if you need a help with and communicate with others. 
+
+## Legends
+
+| <div style="align-text: center;width:200px">Diagram (e.g)</div> | Description |   
+|----------|:-------------:|
+| {{<mermaid>}}
+flowchart TB
+   classDef head fill:#ffbfcf;
+   classDef installed fill:#34ebba;
+   
+   ID(v0.0.1):::head
+{{</mermaid>}} | Operator bundle version which is the head of the channel |
+| {{<mermaid>}}
+flowchart TB
+   classDef head fill:#ffbfcf;
+   classDef installed fill:#34ebba;
+   
+   ID(v0.0.1):::installed
+{{</mermaid>}} | Operator bundle version which is installed currently |
+| {{<mermaid>}}
+flowchart TB
+   classDef head fill:#ffbfcf;
+   classDef installed fill:#34ebba;
+   
+   ID(v0.0.1)
+{{</mermaid>}} | Operator bundle version which is installable |
+| {{<mermaid>}}
+flowchart TB
+   classDef head fill:#ffbfcf;
+   classDef installed fill:#34ebba;
+   
+   subgraph preview
+      ID(v0.0.1) 
+   end
+{{</mermaid>}} | Operator bundle channel. (eg. from the bundle `annotations.yaml`, `operators.operatorframework.io.bundle.channels.v1: preview` ) |
+| {{<mermaid>}}
+flowchart TB
+   classDef head fill:#ffbfcf;
+   classDef installed fill:#34ebba;
+   
+   subgraph "preview (default)"
+      ID(v0.0.1) --> ID2(v0.0.2)
+   end
+{{</mermaid>}} | Operator default channel. (eg. from the bundle `annotations.yaml`, `operators.operatorframework.io.bundle.channel.default.v1: stable` ) |
+| {{<mermaid>}}
+flowchart TB
+   classDef head fill:#ffbfcf;
+   classDef installed fill:#34ebba;
+   
+   A(v0.0.1) --> B(v0.0.4)
+{{</mermaid>}} | An upgrade path to replace one operator bundle version for another using the [CSV][csv-definition] `replaces spec. (eg.`spec.replaces: exampleoperator.v0.0.1`). More info: [here][upgrade-path-replaces]. |
```

Comment:
Same comment here about the CSV. Is it not sufficient to say `An edge from one bundle to another depicts an upgrade path from the bundle with the outgoing edge to the bundle with the incoming edge` 

Basically I'm trying to see if we can talk about these details in this doc just in terms of a graph data structure, so that anyone with a basic understanding of what a graph/tree is can understand this doc without needing any prior knowledge of OLM internals.

Reply:
Regarding this one. My goal/motivation here is the person check the image/diagram and know how to set up the project directly. Because of this, I’d like to stick with the examples.